src/backend/distributed/utils/shard_utils.c

```diff
@@ -55,3 +56,23 @@ GetLongestShardName(Oid citusTableOid, char *finalRelationName)
 
 	return longestShardName;
 }
+
+
+/*
+ * GetLongestShardName is a utility function that creates a hypothetical shard
+ * name for a table that is not distributed yet.
+ *
+ * There is no easy way to create the names of shards before those shards are created
+ * because sequence manipulation in PostgreSQL is not transactional. Therefore we assume
+ * that the shardId values of a partition and those of its parent will be adjacent.
+ */
+char *
+GetLongestHypotheticalShardName(Oid parentTableOid, char *relationName)
+{
+	char *longestShardName = pstrdup(relationName);
+	ShardInterval *shardInterval = LoadShardIntervalWithLongestShardName(parentTableOid);
+	uint64 newShardId = shardInterval->shardId + ShardCount;
```

Comment:
Should we use 
```
CitusTableCacheEntry *cache = GetCitusTableCacheEntry(relationId);
cache->shardIntervalArrayLength
```
Instead of `ShardCount`. Because we might be dealing with a table that has a different shard count?